packages/shared/selectors/k8s.ts

```diff
@@ -0,0 +1,83 @@
+import { K8sResourceCommon } from "@openshift-console/dynamic-plugin-sdk";
+import * as _ from 'lodash';
+import { nsSpecProvider, nsSpecType } from "../../constants";
+import { GetAPIVersionForModel } from "../../types";
+
+type GetStringProperty<T = K8sResourceCommon> = (resource: T) => string;
+
+export const getName: GetStringProperty = (resource) =>
+  resource?.metadata?.name;
+
+export type PersistentVolumeClaimKind = K8sResourceCommon & {
+  spec: {
+    accessModes: string[];
+    resources: {
+      requests: {
+        storage: string;
+      };
+    };
+    storageClassName: string;
+    volumeMode?: string;
+    /* Parameters in a cloned PVC */
+    dataSource?: {
+      name: string;
+      kind: string;
+      apiGroup: string;
+    };
+    /**/
+  };
+  status?: {
+    phase: string;
+  };
+};
+
+export type SecretKind = {
+  data?: { [key: string]: string };
+  stringData?: { [key: string]: string };
+  type?: string;
+} & K8sResourceCommon;
+
+export type NamespaceStoreKind = K8sResourceCommon & {
+  spec: {
+    [key in nsSpecProvider]: {
+      [key: string]: string;
+    };
+  } & {
+    type: nsSpecType;
+  };
+};
+
```

Comment:
these are types not selectors, need to go under types directory...

Reply:
Done moved to type under k8s.ts